auth/auth.go

```diff
@@ -223,13 +224,47 @@ func (auth *Authenticator) getPrincipal(docID string, factory func() Principal)
 	return princ, nil
 }
 
+func (auth *Authenticator) channelCount(princ Principal, user User, roles []Role, scope, collection string) int {
+	cumulativeChannels := 0
+
+	channels := princ.CollectionChannels(scope, collection).Copy()
```

Comment:
why does this need a copy but `role.CollectionChannels(scope, collection)` doesn't need a copy?